skore/src/skore/persistence/item/cross_validation_reporter_item.py

```diff
@@ -0,0 +1,296 @@
+        param_is_default: bool = (
+            k in default_estimator_params and default_estimator_params[k] == v
+        )
+        final_estimator_params[str(k)] = {
+            "value": repr(v),
+            "default": param_is_default,
+        }
+
+    return {
+        "name": name,
+        "module": module,
+        "params": final_estimator_params,
+    }
+
+
+class CrossValidationReporterItem(Item):
+    """Class to persist the reporter of cross-validation."""
+
+    def __init__(
+        self,
+        reporter_bytes: bytes,
+        created_at: Optional[str] = None,
+        updated_at: Optional[str] = None,
+        note: Optional[str] = None,
+    ):
+        """
+        Initialize a CrossValidationReporterItem.
+
+        Parameters
+        ----------
+        reporter_bytes : bytes
+            The raw bytes of the reporter pickle representation.
+        created_at : str, optional
+            The creation timestamp in ISO format.
+        updated_at : str, optional
+            The last update timestamp in ISO format.
+        note : str, optional
+            A note.
+        """
+        super().__init__(created_at, updated_at, note)
+
+        self.reporter_bytes = reporter_bytes
+
+    @classmethod
+    def factory(
+        cls,
+        reporter: CrossValidationReporter,
+        /,
+        **kwargs,
+    ) -> CrossValidationReporterItem:
+        """
+        Create a CrossValidationReporterItem instance from a CrossValidationReporter.
+
+        Parameters
+        ----------
+        reporter : CrossValidationReporter
+
+        Returns
+        -------
+        CrossValidationReporterItem
+            A new CrossValidationReporterItem instance.
+        """
+        if not isinstance(reporter, CrossValidationReporter):
+            raise ItemTypeError(f"Type '{reporter.__class__}' is not supported.")
+
+        return cls(pickle.dumps(reporter), **kwargs)
+
+    @property
+    def reporter(self) -> CrossValidationReporter:
+        """The CrossValidationReporter from the persistence."""
+        return pickle.loads(self.reporter_bytes)
+
+    def as_serializable_dict(self):
+        """Get a serializable dict from the item."""
+        # Get tabular results (the cv results in a dataframe-like structure)
+        cv_results = {
+            key: value.tolist()
+            for key, value in self.reporter._cv_results.items()
+            if key not in ("estimator", "indices") and isinstance(value, numpy.ndarray)
+        }
+
+        metrics_names = list(cv_results)
+        tabular_results = [
+            {
+                "name": "Cross validation results",
+                "columns": metrics_names,
+                "data": list(zip(*cv_results.values())),
+                "favorability": [_metric_favorability(m) for m in metrics_names],
+            }
+        ]
+
+        # Get scalar results (summary statistics of the cv results)
+        scalar_results = [
+            {
```

Comment:
Here, we still transform to a list because we are going to use JSON serialization later?

Reply:
The role of `as_serializable_dict` is to convert the item to a JSON serializable dict -- I don't know if that answers your question?

Reply:
OK it is answering my question that we still use JSON.

Ultimately in the future, those intermediate structure should be serialize in their native style (numpy should be serialized with `np.save`) without going to Python list.

But it is for later.

Reply:
Just to be clear: the JSON stuff produced by `as_serializable_dict` which you are citing here is only for when we send to the frontend.

On the other hand, the logic for saving numpy arrays into skore is the responsibility of `NumpyArrayItem`. Right now `NumpyArrayItem` is also using JSON, but yes, it could definitely be made more efficient.

Reply:
See #1159